components/public-api/gitpod/experimental/v1/oidc.proto

```diff
@@ -0,0 +1,146 @@
+syntax = "proto3";
+
+package gitpod.experimental.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1";
+
+import "google/protobuf/timestamp.proto";
+
+import "gitpod/experimental/v1/pagination.proto";
+
+// Configuration of an OpenID provider instance.
+//
+// For the metadata describing the configuration of OIDC providers, cf.
+// https://openid.net/specs/openid-connect-discovery-1_0.html
+message OIDCProviderConfig {
+  // ID is the unique identifier for the OIDC Config.
+  // Read only.
+  string id = 1;
+
+  // BEGIN – OIDC
+
+  // URL using the https scheme with no query or fragment component that the OP
+  // asserts as its Issuer Identifier. Required.
+  string issuer = 2;
+
+  // A KeySet that can validate the id_token (JSON web token)
+  // Either one is required.
+  string jwks = 3;
+  string jwks_url = 4;
+
+  // Control options for the consent screen.
+  // Optional.
+  string prompt = 5;
+  // A hint to pre-select the tenant from an AD.
+  // Optional.
+  string domain_hint = 6;
+  // Optional.
+  string login_hint = 7;
+
+  // If set, the default claim mapping is overriden.
+  // Optional.
+  bool override_claim_mapping = 8;
+  // Optional.
+  string claim_email_key = 9;
+  // Optional.
+  string claim_groups_key = 10;
+  // Optional.
+  string claim_username_key = 11;
+  // END – OIDC
+
+  // BEGIN – OAuth2 configuration
+  // Required.
+  string client_id = 12;
+  // Required.
+  string client_secret = 13;
+  // Required.
+  string authorization_endpoint = 14;
+  // Required.
+  string token_endpoint = 15;
+  // Required.
+  repeated string scopes_supported = 16;
+
+  // Source for additional claims for the token.
+  // Required.
+  string userinfo_endpoint = 17;
+  // Optional.
+  string userinfo_id_key = 18;
+  // Optional.
+  string userinfo_name_key = 19;
+
+  // Based on this flag, the provider is not using OpenID for authentication,
+  // the request_type will be "code" instead of "id_token", and the
+  // "userinfo_endpoint" will be queried to obtain a profile by default.
+  bool is_oauth_only = 20;
+  // END – OAuth2 configuration
+
+  // List of the JWS signing algorithms (alg values) supported by the OP for the
+  // ID Token to encode the Claims in a JWT. The algorithm RS256 MUST be
+  // included. Optional.
+  repeated string id_token_signing_alg_values_supported = 21;
+
+  // Time when the config was created.
+  // Read-only.
+  google.protobuf.Timestamp creation_time = 22;
+}
+
+service OIDCProviderConfigService {
```

Comment:
Could we simply this to `OIDCService`, alternatively, `OIDCSProviderService`? Config is effectively what we're putting in the resource so I believe we don't need it in the service title.

We can then also simplify the RPCs to drop the `Config` part

Reply:
Thought about that.

Let's have a look at an example which shows the issue with simplifying the name. 

The *OIDC provider* (IdP) may be Google
   * the metadata for this provider are declared https://accounts.google.com/.well-known/openid-configuration
   * that provider supports multiple clients, a single Gitpod installation may have configured several clients at the same time
   * (in fact) the proper term would be "OIDC client configuration"
   * leaving out the configuration removes the necessary additional dimension

Reply:
So the Resource is OIDC Client Configuration, so the underlying object we store (and type here can remain `OIDCClientConfiguration` but the service doesn't need encode the full resource. The service, and the RPC for a unique identifier. So together, we can have

`OIDCService.CreateClientConfiguration()` for example